[zh-CN]: Translate "Forms and buttons in HTML"

NaomiZhu23 opened the pull request:
### Description

🚀 Complete Simplified Chinese (zh-CN) translation of the entire page:
“Forms and buttons in HTML” (/Learn_web_development/Core/Structuring_content/HTML_forms).

### Motivation

📚 While relearning HTML basics, I noticed this page has no Chinese version. A full translation will help Chinese-speaking readers better understand HTML forms and buttons, especially those less comfortable with English documentation. ✅

🍀 **Note**: This is my first contribution; any feedback is highly appreciated.

### Additional details

None.

### Related issues and pull requests

None.

NaomiZhu23 committed:
[zh-CN]: translate Forms and Buttons in HTML.

NaomiZhu23 committed:
Resolved several translation errors.

NaomiZhu23 committed:
[zh-CN]: Resolved several translation errors.

NaomiZhu23 committed:
Add necessary spaces.

NaomiZhu23 committed:
Merge remote-tracking branch 'upstream/main' into zh-CN/HTML-Forms-and-Buttons
Merge upstream changes into zh-CN/HTML-Forms-and-Buttons branch.

NaomiZhu23 committed:
Merge branch 'main' into zh-CN/HTML-Forms-and-Buttons

NaomiZhu23 committed:
Merge branch 'main' into zh-CN/HTML-Forms-and-Buttons

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+`<label>` 元素的重要性体现在多个方面，尤其包括以下几点：
+
+- 当视力障碍用户使用屏幕阅读器浏览网页时，屏幕阅读器会在遇到每个控件时朗读其关联的标签文本，从而帮助用户理解每个控件需要填写什么内容。
+- 用户可以通过点击标签文本来聚焦表单控件，而不仅限于点击控件本身。这对移动设备用户尤为重要——在触摸屏上精确点击小控件可能比较困难，而增大 **点击区域** 有助于提高可用性。
```
与原文有点出入

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+#### 显式与隐式表单标签
+
+你在上文中看到的标签样式被称为 **显式表单标签**（explicit form label）——控件与标签之间的关联是通过 `id` 和 `for` 属性显式建立的。你也可以通过将控件嵌套在标签内的方式实现 **隐式表单标签**（implicit form label），如下所示：
```
```suggestion
你在上文中看到的标签样式被称为**显式表单标签**——控件与标签之间的关联是通过 `id` 和 `for` 属性显式建立的。你也可以通过将控件嵌套在标签内的方式实现 **隐式表单标签**，如下所示：
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+通过嵌套方式，控件与标签之间的关联是隐式建立的，因此你不再需要使用 `id` 和 `for` 属性。
+
+这两种方式都可以使用，但我们建议你使用显式标签的方式。因为显式标签关联更容易识别和理解，尤其当你的 HTML 代码变得复杂时。此外，屏幕阅读器（以及其他辅助技术）有时不能正常处理隐形标签。
```
```suggestion
这两种方式都可以使用，但我们建议你使用显式标签的方式。因为显式标签关联更容易识别和理解，尤其当你的 HTML 代码变得复杂时。此外，屏幕阅读器（以及其他辅助技术）有时不能正常处理隐式标签。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+通过 `<button>` 元素的 `type` 属性可以指定按钮的其他行为：
+
+- `<button type="submit">` 明确声明该按钮是“提交按钮”。一般来说不需要显式声明，除非你的 `<form>` 中包含了多个按钮，并且你想明确表示哪个按钮是用于提交表单的。这种情况非常罕见。
+- `<button type="reset">` 创建一个 _重置按钮_ ——这会立即清空表单中输入的所有内容，将表单恢复到初始状态。 **不建议使用重置按钮** ——在早期的网页中它们比较常见，但通常会造成更多困扰。许多人曾经经历过填写完长长的表单后，不小心点击了“重置”按钮而不是“提交”，结果只能重新填写表单。
```
```suggestion
- `<button type="reset">` 创建一个*重置按钮*——这会立即清空表单中输入的所有内容，将表单恢复到初始状态。**不建议使用重置按钮**——在早期的网页中它们比较常见，但通常会造成更多困扰。许多人曾经经历过填写完长长的表单后，不小心点击了“重置”按钮而不是“提交”，结果只能重新填写表单。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+> [!NOTE]
+> 你也可以通过 `<input>` 元素配合不同的 `type` 属性来创建上述三种按钮：[`<input type="submit">`](/zh-CN/docs/Web/HTML/Reference/Elements/input/submit)、[`<input type="reset">`](/zh-CN/docs/Web/HTML/Reference/Elements/input/reset) 和 [`<input type="button">`](/zh-CN/docs/Web/HTML/Reference/Elements/input/button)。但与 `<button>` 相比，它们存在许多限制，因此推荐使用 `<button>` 元素。
+
+## 关于可访问性的小提示
```
accessibility → 无障碍（性），下同

```suggestion
## 关于无障碍的小提示
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+## 关于可访问性的小提示
+
+我们之前已经提到过表单标签对于可访问性的重要性，但我们还想补充一些关于使用语义化 HTML 元素创建表单的重要性的观点。（例如，使用 `<button>` 提交表单，而不是使用一个通过 CSS 和 JavaScript 模拟出来的 `<div>` 元素）。你确实可以利用 CSS 和 JavaScript 把几乎任何 HTML 元素“伪装成”表单控件，看起来像是一个按钮、输入框、下拉菜单等。开发者这么做通常是出于设计上的考虑——某些原生表单控件确实比较难自定义样式。
```
_correct_ 没译出来

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+- 语义化元素可以被屏幕阅读器等辅助技术识别，并将内容传递给视觉障碍的用户。
+- 表单控件和按钮默认支持键盘访问。在先前的示例中，你可以尝试使用 <kbd>Tab</kbd> 和 <kbd>Shift</kbd> + <kbd>Tab</kbd> 切换表单元素（这种操作被称为“Tab 键导航”）。
+- 你还会注意到，当你使用 Tab 在表单元素间进行切换时，当前聚焦的元素会被蓝色轮廓标记出来（称为 **焦点轮廓**）。这对于使用键盘进行操作的用户来说非常重要，因为它帮助用户知道自己目前操作的是哪一项。
```
粗斜体样式不需要留空格

```suggestion
- 你还会注意到，当你使用 Tab 在表单元素间进行切换时，当前聚焦的元素会被蓝色轮廓标记出来（称为**焦点轮廓**）。这对于使用键盘进行操作的用户来说非常重要，因为它帮助用户知道自己目前操作的是哪一项。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+            name="hotel"
+            value="economy"
+            checked />
+          <label for="hotelChoice1">经济型 (+$0)</label>
```
换成中文括号

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+“选择酒店房型”按钮是通过 [`<input type="radio">`](/zh-CN/docs/Web/HTML/Reference/Elements/input/radio) 控件实现的。这些控件会渲染为一组按钮控件，每次只能选择其中一个——你不能同时选中多个。它们的名字来源于老式收音机上的按钮：按下一个按钮，之前选中的按钮就会弹出。
+
+我们的示例代码如下
```
```suggestion
我们的示例代码如下：
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+`radio` 输入控件的工作方式大致和 `text` 输入控件相同，但也有几点不同：
+
+- 每组单选按钮的 `name` 属性值必须相同，以便将它们关联成一组。如果 `name` 属性值不同，它们会被视为不同的组，提交时也会有发送不同的值。
+- 每个单选按钮必须包含一个 `value` 属性，用于指定提交时该按钮对应的值。提交的数据会是一个名值对（name/value pair），其中名称（即 `name` ）始终相同，例如 `hotel=economy` 或 `hotel=superior` 。
```
```suggestion
- 每个单选按钮必须包含一个 `value` 属性，用于指定提交时该按钮对应的值。提交的数据会是一个名值对，其中名称（即 `name`）始终相同，例如 `hotel=economy` 或 `hotel=superior`。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+- 每个单选按钮的 `<label>` 应该描述该具体选项的值，而不是你正在选择的这一组选项所代表的字段含义。描述整组选项的推荐做法是使用 {{htmlelement("fieldset")}} 元素包裹这些选项，并用其子元素 {{htmlelement("legend")}} 来提供整体描述。
+
+> [!NOTE]
+> 除了用于结构化和标记表单外，fieldset 还有其他用途，例如可用于一次性 [禁用](#disabling_form_controls) 整组控件
```
注意锚点链接
```suggestion
> 除了用于结构化和标记表单外，fieldset 还有其他用途，例如可用于一次性[禁用](#禁用表单控件)整组控件。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+> [!NOTE]
+> 除了用于结构化和标记表单外，fieldset 还有其他用途，例如可用于一次性 [禁用](#disabling_form_controls) 整组控件
+
+还值得注意的是，我们为第一个单选按钮添加了 `checked` 属性 —— 这会让它在页面初次加载时默认被选中。这意味着总会有一个选项被选中，并且你无法取消选中一个单选按钮，除非选择另一个。
```
```suggestion
还值得注意的是，我们为第一个单选按钮添加了 `checked` 属性——这会让它在页面初次加载时默认被选中。这意味着总会有一个选项被选中，并且你无法取消选中一个单选按钮，除非选择另一个。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
````diff
@@ -0,0 +1,505 @@
+
+```html
+<fieldset>
+  <legend>要参加的课程：</legend>
````
```suggestion
  <legend>选择要参加的课程：</legend>
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+正如你在代码片段中看到的，单选按钮和复选框的实现方式非常相似（它们都可以使用 `checked` 属性，在页面加载时默认被选中）。它们的行为也比较类似，不同点在于单选按钮允许你从多个选项中选择零个或一个，而复选框允许你从多个选项中选择零个或多个。
+
+主要区别（除了 `type` 属性值的不同）在于每个复选框的 `name` 属性值不同，且通常没有设置 `value` 属性。这意味着，每个复选框代表不同的数据值，而单选按钮只能代表一个值。当表单提交时，如果复选框被选中，则提交的值为 `on`，例如： `yoga=on`、`ballon=on`等等。
```
```suggestion
主要区别（除了 `type` 属性值的不同）在于每个复选框的 `name` 属性值不同，且通常没有设置 `value` 属性。这意味着，每个复选框代表不同的数据值，而单选按钮只能代表一个值。当表单提交时，如果复选框被选中，则提交的值为 `on`，例如：`yoga=on`、`ballon=on` 等等。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+- 客户端验证：在浏览器端进行，通常通过表单验证属性（如 `required`）和 JavaScript 结合实现。客户端验证可以即时提示用户输入有误，但它并不可靠，不能有效阻止恶意数据。因为用户可以关闭 JavaScript 或修改客户端代码，从而绕过验证。
+- 服务器端验证：在服务器上进行，使用服务器端所采用的任意语言实现。错误格式的数据可以是意外发送的，也可以是有意而为之。经验法则是：服务器永远不要信任客户端发送的任何数据，以避免由于格式错误的消息而引发漏洞或安全问题。服务器端验证更适合阻止恶意消息，因为服务器端代码更难被篡改。但它无法像客户端验证那样及时给用户反馈，因为数据必须先发送到服务器验证，服务器再把结果返回给用户。
+
+简而言之，不要只选择客户端验证或服务器端验证——你需要两者结合。你需要客户端验证来为用户的输入提供即时反馈；同时也需要服务器端验证，来确保发送到服务器的消息格式是安全、可处理的。如果你想开始深入学习表单验证，可以从 [客户端表单验证](/zh-CN/docs/Learn_web_development/Extensions/Forms/Form_validation) 开始。
```
```suggestion
简而言之，不要只选择客户端验证或服务器端验证——你需要两者结合。你需要客户端验证来为用户的输入提供即时反馈；同时也需要服务器端验证，来确保发送到服务器的消息格式是安全、可处理的。如果你想开始深入学习表单验证，可以从[客户端表单验证](/zh-CN/docs/Learn_web_development/Extensions/Forms/Form_validation)开始。
```

jasonren0403 replied:
```suggestion
简而言之，不要只选择客户端验证或服务器端验证——你需要两者结合。你需要客户端验证来为用户的输入提供即时反馈；同时也需要服务器端验证，来确保发送到服务器的消息格式是安全、可处理的。如果你想开始深入学习表单验证，可以从[客户端表单验证](/zh-CN/docs/Learn_web_development/Extensions/Forms/Form_validation)开始。
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+## 技能测试！
+
+你已经阅读完本文，但你能记住重要的信息吗？在继续之前，你可以进行一些进一步的测试，以验证你是否掌握了这些信息。请参阅 [技能测试：表单和按钮](/zh-CN/docs/Learn_web_development/Core/Structuring_content/Test_your_skills/Forms_and_buttons)
```
```suggestion
你已经阅读完本文，但你能记住重要的信息吗？在继续之前，你可以进行一些进一步的测试，以验证你是否掌握了这些信息。请参阅[技能测试：表单和按钮](/zh-CN/docs/Learn_web_development/Core/Structuring_content/Test_your_skills/Forms_and_buttons)
```

jasonren0403 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+## 参见
+
+- [Web 表单 —— 处理用户数据](/zh-CN/docs/Learn_web_development/Extensions/Forms)
```
```suggestion
- [Web 表单——处理用户数据](/zh-CN/docs/Learn_web_development/Extensions/Forms)
```

NaomiZhu23 commented on files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md:
```diff
@@ -0,0 +1,505 @@
+
+如果你立刻点击“订阅新闻简报”按钮，会看见一个验证错误，因为没有填写任何数据。如果你填写了姓名和邮箱地址后再点击该按钮，则会看到一个 `404` 错误页面。
+
+我们稍后会解释原因。在此之前，请将上面的 HTML 代码复制到你的 [代码编辑器](/zh-CN/docs/Learn_web_development/Getting_started/Environment_setup/Code_editors) 新建的 HTML 文件中，然后在新的浏览器标签页打开它。
```
好的！

NaomiZhu23 committed:
Update files/zh-cn/learn_web_development/core/structuring_content/html_forms/index.md

Applied some of the suggested changes; will address the remaining updates manually later.

Co-authored-by: Jason Ren <40999116+jasonren0403@users.noreply.github.com>

NaomiZhu23 committed:
apply suggestions

NaomiZhu23 committed:
Merge branch 'main' into zh-CN/HTML-Forms-and-Buttons

NaomiZhu23 committed:
Remove unnecessary spacing.

NaomiZhu23 committed:
update some text.

NaomiZhu23 committed:
fix an issue

NaomiZhu23 committed:
Apply suggestions

Co-authored-by: Jason Ren <40999116+jasonren0403@users.noreply.github.com>

NaomiZhu23 committed:
Address the requested changes

Address the requested changes from the earlier review.

Co-authored-by: Jason Ren <40999116+jasonren0403@users.noreply.github.com>

jasonren0403 committed:
fixup

jasonren0403 reviewed (approved):
👍

NaomiZhu23 commented:
> Thanks for your contribution ❤️ @NaomiZhu23

Thank you for your guidance! ❤️ Appreciate a lot!
